### DISCUSSION TITLE
Add IOS Navigation Gestures

### DISCUSSION THREAD

**Discussion by ghost** [home-assistant/frontend#11198] [Other feature requests]
*[move[bot]](https://github.com/apps/move) commented on Nov 30, 2018, 6:49 PM UTC:*

_[NovapaX](https://github.com/NovapaX) commented on Feb 26, 2018, 12:03 PM UTC:_

In iOS safari:

*   Swipe from the edge of the screen is the back-gesture. So it does not work to open the menu.
*   Need scroll all the way to the top of the page to 'uncondense' the header to show the menu button.

When adding hass as a web-app to the homescreen on iOS:

*   Swipe to open the menu works, but the back-gesture is gone and no navigation buttons are displayed. So the only way to close the more-info is to reach all the way up.

_This issue was moved by [zsarnett](https://github.com/zsarnett) from [home-assistant/home-assistant-polymer#931](https://github.com/home-assistant/home-assistant-polymer/issues/931)._

*This issue was moved by [iantrich](https://github.com/iantrich) from [home-assistant/ui-schema#204](https://github.com/home-assistant/ui-schema/issues/204).*

**Comment by ghost** [comment]
*[move[bot]](https://github.com/apps/move) commented on Nov 30, 2018, 6:49 PM UTC:*

_[balloob](https://github.com/balloob) commented on Feb 26, 2018, 7:50 PM UTC:_

If you have multiple views, you can tap the current tab to scroll to the top.

**Comment by ghost** [comment]
*[move[bot]](https://github.com/apps/move) commented on Nov 30, 2018, 6:49 PM UTC:*

_[balloob](https://github.com/balloob) commented on Feb 26, 2018, 7:50 PM UTC:_

I also thought that iOS is very heavy on having close buttons on the top of the screen?

**Comment by ghost** [comment]
*[move[bot]](https://github.com/apps/move) commented on Nov 30, 2018, 6:49 PM UTC:*

_[NovapaX](https://github.com/NovapaX) commented on Feb 27, 2018, 9:13 AM UTC:_

Yes, but therefore it is also quite heavy on the gestures.
But maybe the last point is the 'worst'. (closing the more-info is a frequent action), and without the back gesture or a dismiss or back button at the bottom it is a far reach.

Will do some research, but as this is pretty low priority I'll probably revisit this once iOS 11.3 comes out and do an optimization to PWA and webapp-manifest... (not planning on doing testing and optimization twice in a short period ;)
see: [https://medium.com/@firt/pwas-are-coming-to-ios-11-3-cupertino-we-have-a-problem-2ff49fd7d6ea](https://medium.com/@firt/pwas-are-coming-to-ios-11-3-cupertino-we-have-a-problem-2ff49fd7d6ea)

**Comment by ghost** [comment]
*[move[bot]](https://github.com/apps/move) commented on Nov 30, 2018, 6:49 PM UTC:*

_[balloob](https://github.com/balloob) commented on Feb 27, 2018, 9:32 AM UTC:_

Actually, I've included a "back" hack so that you can swipe from the left to go back in the browser to close the more info. That's why the Android back button works <g-emoji class="g-emoji" alias="+1" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f44d.png">👍</g-emoji>

**Comment by ghost** [comment]
*[move[bot]](https://github.com/apps/move) commented on Nov 30, 2018, 6:49 PM UTC:*

_[NovapaX](https://github.com/NovapaX) commented on Feb 27, 2018, 9:34 AM UTC:_

yes, I know... But the navigation gestures do not work when it is added as a webapp in 'standalone' mode (so without the safari interface elements)

**Comment by bramkragten** [comment]
Is this still an issue with the iOS app?
